completion/available/system.completion.bash

```diff
@@ -3,26 +3,43 @@
 # Loads the system's Bash completion modules.
 # If Homebrew is installed (OS X), its Bash completion modules are loaded.
 
-if [ -f /etc/bash_completion ]; then
-  . /etc/bash_completion
+if [ -r /etc/bash_completion ] ; then
+  # shellcheck disable=SC1091
+  source /etc/bash_completion
 fi
 
 # Some distribution makes use of a profile.d script to import completion.
-if [ -f /etc/profile.d/bash_completion.sh ]; then
-  . /etc/profile.d/bash_completion.sh
+if [ -r /etc/profile.d/bash_completion.sh ] ; then
+  # shellcheck disable=SC1091
+  source /etc/profile.d/bash_completion.sh
 fi
 
 
-if [ $(uname) = "Darwin" ] && command -v brew &>/dev/null ; then
-  BREW_PREFIX=$(brew --prefix)
+if [ "$(uname)" == "Darwin" ] ; then
 
-  if [ -f "$BREW_PREFIX"/etc/bash_completion ]; then
-    . "$BREW_PREFIX"/etc/bash_completion
+  # XCode installs git's completion files to a non-standard place
+  # Note: This should be loaded before brew's completion folder is parsed,
+  #       multiple tools glob on to this completion.
+  _git_comp='/Applications/Xcode.app/Contents/Developer/usr/share/git-core/git-completion.bash'
+  if _command_exists git && ! complete -p git &>/dev/null && [ -r "$_git_comp" ] ; then
+    # shellcheck disable=SC1090
+    source "$_git_comp"
   fi
+  unset _git_comp
 
- # homebrew/versions/bash-completion2 (required for projects.completion.bash) is installed to this path
-  if [ "${BASH_VERSINFO}" -ge 4 ] && [ -f "$BREW_PREFIX"/share/bash-completion/bash_completion ]; then
-    export BASH_COMPLETION_COMPAT_DIR="$BREW_PREFIX"/etc/bash_completion.d
-    . "$BREW_PREFIX"/share/bash-completion/bash_completion
+  if _command_exists brew ; then
+    BREW_PREFIX=$(brew --prefix)
+
+    if [ -r "$BREW_PREFIX"/etc/bash_completion ] ; then
+      # shellcheck disable=SC1090
+      source "$BREW_PREFIX"/etc/bash_completion
+    fi
+
+    # homebrew/versions/bash-completion2 (required for projects.completion.bash) is installed to this path
+    if [ -r "$BREW_PREFIX"/etc/profile.d/bash_completion.sh ] ; then
```

Comment:
What happened to the `$BREW_PREFIX/share/bash_completion` location? I still have that directory, but the script does not seem to be there anymore. Has that been moved to `/etc/profile.d`?

Reply:
@nwinkler Yeah it appears that they've moved it. Your note about using `bash-completion@2` got me to test that, and if you run `brew info bash-completion@2` you'll see that they've updated the instructions to point at this new location, which in turn has more-or-less the same checks you were doing.